java/arcs/core/util/Result.kt

```diff
@@ -30,3 +30,28 @@ suspend fun <T> resultOfSuspend(block: suspend () -> T): Result<T> = try {
 } catch (e: Throwable) {
     Result.Err(e)
 }
+
+/**
+ * If this is [Result.Err], rethrows the wrapped exception. Otherwise, returns the wrapped value.
+ *
+ * This extension method can be used with [arcs.core.util.resultOf] to short-circuit the execution
+ * of a block of code. e.g.,
+ *
+ * resultOf {
+ *     val resultX = ...
+ *     val resultY = ...
+ *     val x = resultX.getOrThrow() // if resultX is [Result.err], we break control here.
+ *     val y = resultY.getOrThrow()
+ *     val z = foo(x, y) // x and y are unwrapped here.
+ * }
+ */
+fun <T> Result<T>.getOrThrow(): T = when (this) {
```

Comment:
Does this need to be an extension function? It looks like you could just move it up to where the class is defined?

Reply:
Good idea. I moved these methods into the `Result<T>` class itself. (I just started it as an extension method in a different package, but left it as is when I moved to the `Result.kt` file, which was not a good thing.)